expression/column.go

```diff
@@ -160,6 +160,10 @@ type Column struct {
 	// Index is used for execution, to tell the column's position in the given row.
 	Index int
 
+	// Flag is a flag comes from model.ColumnInfo. We maintain here mainly because we
+	// want to eliminate ifnull on not null column.
+	Flag uint
```

Comment:
`Column.Flag` and `Column.RetType.Flag` makes the code hard to be maintained and also it's error-prone.

Reply:
> @zz-jason Just tried to maintain Flag in RetType of expression.Column but failed to do so. It seems RetType is a reference of original types.FieldType. If we modify it in place, it causes other test cases failure.

@zhexuany I think you should try to fix the tests by generating a new `FieldType` if necessary.